lapack/testlapack/dlanhs.go

```diff
@@ -0,0 +1,36 @@
+// Copyright ©2023 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"math"
+	"testing"
+
+	"golang.org/x/exp/rand"
+	"gonum.org/v1/gonum/lapack"
+)
+
+type Dlanhser interface {
+	Dlanhs(norm lapack.MatrixNorm, n int, a []float64, lda int, work []float64) (resultNorm float64)
+}
+
+func DlanhsTest(t *testing.T, impl Dlanhser) {
+	const tol = 1e-8
+	work := make([]float64, 9*9)
```

Comment:
Please make this the minimum required length and fill it with garbage before every call to Dlanhs.